### PR TITLE
[Tests] Check QueryProfileShardResult parser robustness for new fields

### PR DESCRIPTION
**Pull request by cbuescher**
When parsing resonses we should be ignoring any new unknown fields or inner objects in most cases to be forward compatible with changes in core on the
client side. This change adds test for this for QueryProfileShardResult and nested substructures and changes the parsing code where necessary to be able to
ignore new fields and objects in the xContent.

### CODE REVIEW

**Review by javanna — APPROVED** [review]
LGTM thanks @cbuescher

File: core/src/main/java/org/elasticsearch/search/profile/SearchProfileShardResults.java [hunk 1]
```diff
@@ -109,14 +106,21 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     public static SearchProfileShardResults fromXContent(XContentParser parser) throws IOException {
         XContentParser.Token token = parser.currentToken();
-        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
         Map<String, ProfileShardResult> searchProfileResults = new HashMap<>();
-        ensureFieldName(parser, parser.nextToken(), SHARDS_FIELD);
-        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser::getTokenLocation);
-        while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-            parseSearchProfileResultsEntry(parser, searchProfileResults);
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
```
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/search/profile/SearchProfileShardResults.java]: nit: we can check the expected token and then create the searchProfileResults map

File: core/src/test/java/org/elasticsearch/search/profile/ProfileResultTests.java [hunk 2]
```diff
@@ -62,12 +64,32 @@ public static ProfileResult createTestItem(int depth) {
     }
 
     public void testFromXContent() throws IOException {
+        dofromXContentTestWithRandomFields(false);
```
Comment (tlrx) [inline core/src/test/java/org/elasticsearch/search/profile/ProfileResultTests.java]: nit: do**F**romXContentTestWithRandomFields

File: core/src/test/java/org/elasticsearch/search/profile/SearchProfileShardResultsTests.java [hunk 3]
```diff
@@ -58,20 +60,43 @@ public static SearchProfileShardResults createTestItem() {
     }
 
     public void testFromXContent() throws IOException {
+        dofromXContentTestWithRandomFields(false);
```
Comment (tlrx) [inline core/src/test/java/org/elasticsearch/search/profile/SearchProfileShardResultsTests.java]: nit: do**F**romXContentTestWithRandomFields

**Review by tlrx — APPROVED** [review]

**Pull request merged**